services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusResponse.kt

```diff
@@ -15,6 +17,8 @@ data class HakemusResponse(
     val applicationType: ApplicationType,
     val applicationData: HakemusDataResponse,
     val hankeTunnus: String,
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    val ilmoitukset: Map<IlmoitusType, List<IlmoitusResponse>>?,
```

Comment:
First I thought that is it a bit redundant to have the type two times in the response structure:

```
  "ilmoitukset": {
    "TOIMINNALLINEN_KUNTO": [
      {
        "type": "TOIMINNALLINEN_KUNTO",
        "dateReported": "2024-08-16",
        "reportedAt": "2024-08-16T12:21:20.347402Z"
      }
    ]
  },
```
But then, maybe it is more user-friendly (UI-friendly) like this.

Reply:
I didn't think about that. I guess it doesn't hurt to repeat the type in each object. This also makes it easier to reuse the DTOs.